### DISCUSSION TITLE
Can't submit a comment

### DISCUSSION THREAD

**Discussion by Wonderfall** [posativ/isso#789] [Q&A]
Hi, thanks for your software.
I got Isso working on my own blog, and it works flawlessly using a [docker container](https://hub.docker.com/r/wonderfall/isso/).

Someone asked me to do a similar blog (Ghost + Isso), so did I (hosted on the same server). But Isso doesn't work though everything should be OK :   whenever I try to submit a comment, nothing happens. I found something, using "Inspect Element" : 

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>URI does not exist</p>
```

Interesting. But why? Since Isso is working on my own blog with the same configuration files, I see no reason that can explain this.

Configuration files ("isso-one" works, "isso-two" does not) : 
- [Simplified docker-compose.yml](https://paste.schrodinger.io/?01a9869a6168ceac#KiDe8NuEB+J6hGrmrH4yXceo2sQEjXsxbDXTS79NWcI=)
- [isso.conf](https://paste.schrodinger.io/?b1494db2e309cedd#QfQFB416h5TdptJ7orpjiR59Se8R2xWHubD+qD9owSc=) from isso-one
- [isso.conf](https://paste.schrodinger.io/?c46bda607b880ad0#p+gaarjM0HGgQnHNJX9jR/ZEFtXi7wBvFywS7a36A2w=) from isso-two
- [nginx vhost](https://paste.schrodinger.io/?6497f55b02787f56#GaqsDxhmR120QAUpWTtyn26ug+2AEDhPtu5Pi3pIm1M=) for isso-one
- [nginx vhost](https://paste.schrodinger.io/?fc893e004c29e37e#3e345NlqHvIeBcuTgtxfDoyAnM76U5Xgjt0J03iZpq8=) for isso-two

Of course I can privately share the true adresses if it can help.
Isso logs show nothing useful, except connection to both SMTP and website is OK.

May you help me? I think I've tried everything I could imagine... Thanks.

**Comment by posativ** [comment]
I don't have a working Docker setup currently, but I might be able to setup a virtual machine for this.

PS: did you create the fancy pokemon image?

**Comment by Wonderfall** [comment]
You can find my Dockerfile [here](https://github.com/Wonderfall/dockerfiles/tree/master/isso), whis is very simple.
(Just in case, I think my attempt to dockerize isso was successful since it works [here](https://cats.schrodinger.io) (I think I could share it before, if you wanna try). You certainly know that two containers (from the same image) should produce the same result, in fact it shouldn't be different for the second instance. On top of that, configuration files are almost identical. That's why I'm getting crazy... Perhaps I missed something?)

PS : Unfortunately, I didn't create this fancy pokémon image (but I found it cute!).

**Comment by baptistedonaux** [comment]
Currently, I have the problem to send comments.

I use the wonderfall/isso Docker image. This is my docker-compose.yml

```
version: '2'

services:
  isso:
    image: wonderfall/isso
    environment:
      - GID=1000
      - UID=1000
    volumes:
      - ./docker/comment/isso.conf:/config/isso.conf
      - ./docker/db:/db
    dns_search: local.comments.www.baptiste-donaux.fr
```

**Comment by baptistedonaux** [comment]
It's a fail which thrown when there are no comment attached at current page.

**Comment by kinosang** [comment]
Have the same issue.

Get `404 Not Found` when submit comments

**Comment by jferrara77** [comment]
I also ran into this sporadic issue that I didn't have in one instance of isso but had in another. I traced the problem to lines 171-175 of the view/comment.py file. Not sure what the intent here exactly is, I'm certainly no python expert, but the attempt to read a title and uri for a thread before the thread has been created seems strange. To get it working, I just commented out these lines and set the thread directly with an 'Untitled' title. But maybe there's a reason for this so I'll leave it someone with more insight to resolve permanently.

**Comment by jGleitz** [comment]
I think the issue is caused by the fact that isso assumes that comments are only posted on real, existing web pages (which usually is the case). The `404` means that isso is not able to locate the web page the comment is to be posted on. 

For now, I’d assume that there’s a problem with your `host` configuration (see [server configuration](https://posativ.org/isso/docs/configuration/server/)). Providing the correct host there might fix the issue.

Note that #252 would fix this issue, too. (Because you’d be able to provide the title yourself and isso wouldn’t require to find the `uri` anymore).

**Comment by jferrara77** [comment]
Thanks for that, I got my problem solved properly now, which has to do with the fact that I'm using isso for comments on a federation of sites and so I have multiple sites' URI's mapped to the same comment thread which breaks this local("origin") based request that produces the title and uri for the thread that's based on the host name set in the config file(which I'm ignoring in my modified version)..

Anyway, thanks again for the help, I got it under control here now.

**Comment by kinosang** [comment]
@jGleitz isso can access to my site, it shows that `INFO: connected to http://def.abc.com/`on startup.

**Comment by jGleitz** [comment]
> @jGleitz isso can access to my site, it shows that `INFO: connected to http://def.abc.com/on` startup.

are you sure that the uri you are providing is correct and points to an existing page of your web presence? Can you post the request that fails?

**Comment by kinosang** [comment]
@jGleitz 
you mean, I should add all the pages into isso.cfg before I use it?

Here's the HTTP request and response

```
http://comment.abc.com/new?uri=%2Ffile%2Fdetails%2F03b89535285f67f29d1a2db5a91e2184ed4d5504
```

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>URI does not exist</p>
```

**Comment by jGleitz** [comment]
> you mean, I should add all the pages into isso.cfg before I use it?

certainly not. 

Another reason I can think of: isso doesn’t seem to follow redirects. So it may be possible that the URI is working in your browser, but not for isso.

I, for example, had to add a slash at the end: `/test/page` did not work, but `/test/page/` did (although both work in the browser).

**Comment by ix5** [comment]
@Wonderfall is this still a current issue?
